datadog-log-ffi/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use datadog_log::logger;
+use datadog_log::logger::{logger_configure, logger_set_log_level};
+use ddcommon_ffi::{CharSlice, Error};
+
+/// Sets the global log level.
+///
+/// # Arguments
+/// * `log_level` - The minimum level for events to be logged
+///
+/// # Errors
+/// Returns an error if the log level cannot be set.
+#[no_mangle]
+pub extern "C" fn ddog_logger_set_log_level(
+    log_level: logger::LogEventLevel,
+) -> Option<Box<Error>> {
+    logger_set_log_level(log_level).err().map(Box::new)
+}
+
+/// Configures the global logger.
+///
+/// # Arguments
+/// * `config` - Configuration for the logger including level and output destination
+///
+/// # Errors
+/// Returns an error if the logger cannot be configured.
+#[no_mangle]
+pub extern "C" fn ddog_logger_configure(config: LoggerConfig) -> Option<Box<Error>> {
```

Comment:
Just wondering if it'd be worth hiding the LoggerConfig and just provide a couple of simple functions to just log to fd(stdout)? Do we plan to support many more methods?

Reply:
The logger config will have more things such as rate limiting interval, so ideally we want to have a config type for better forward compatibility. Thinking on having each logging type their own methods we have to do something like
- `ddog_logger_disable()`
- `ddog_logger_configure_std(config: StdConfig)`
- `ddog_logger_configure_file(config: FileConfig)`

tbh I'm fine with either since these all have similar limitations.

What I would definitely avoid is passing n-different params for each config which is going to be a bad API from compatibility point of view.

Reply:
I'm in favor of following a constructor pattern or break down the underlying logic into smaller functions. In this case I like the approach you suggested, that way we can isolate the inner workings of the configuration and there is less cognitive load for the library user. I agree that a struct provides better encapsulation and forward compatibility.